### PR TITLE
Remove aws-parallelcluster-common cookbook

### PR DESCRIPTION
**Pull request by enrico-usai**
### Description of changes
common/libraries used by tests (instance.rb, node.rb, os_properties.rb) have been moved to shared

### CODE REVIEW

File: .github/workflows/dokken-unit-tests.yml [hunk 1]
```diff
@@ -3,60 +3,6 @@ name: ParallelCluster Cookbook Unit Tests on Docker
 on: [workflow_dispatch]
```
Comment (awslitvit) [inline .github/workflows/dokken-unit-tests.yml:3]: I wonder if we still need this workflow at all. We never used it and often forgot to maintain.

File: cookbooks/aws-parallelcluster-computefleet/test/inspec.yml [hunk 2]
```diff
@@ -4,4 +4,4 @@ license: 'Apache-2.0'
 
 depends:
   - name: common
-    path: cookbooks/aws-parallelcluster-common/test/common
```
Comment (awslitvit) [inline cookbooks/aws-parallelcluster-computefleet/test/inspec.yml:7]: Are we sure common isn't used by ImageBuilder?

File: cookbooks/aws-parallelcluster-environment/spec/unit/libraries/volume_spec.rb [hunk 3]
```diff
@@ -1,14 +1,6 @@
 require 'spec_helper'
 
 describe 'aws-parallelcluster-environment:libraries:format_directory' do
-  cached(:chef_run) do
```
Comment (awslitvit) [inline cookbooks/aws-parallelcluster-environment/spec/unit/libraries/volume_spec.rb:4]: Love it! In the future, we can test other library functions as well.

File: util/bump-version.sh [hunk 4]
```diff
@@ -29,25 +29,24 @@ sed -i "s/version '${CURRENT_PCLUSTER_VERSION_SHORT}'/version '${NEW_PCLUSTER_VE
 sed -i "s/ENV\['KITCHEN_PCLUSTER_VERSION'\] || '${CURRENT_PCLUSTER_VERSION}'/ENV\['KITCHEN_PCLUSTER_VERSION'\] || '${NEW_PCLUSTER_VERSION}'/g" kitchen.ec2.yml
 
 # Update dependencies version in main cookbook metadata
-COOKBOOKS=("aws-parallelcluster-common" "aws-parallelcluster-awsbatch" "aws-parallelcluster-config" "aws-parallelcluster-entrypoints" "aws-parallelcluster-scheduler-plugin" "aws-parallelcluster-slurm" "aws-parallelcluster-test" "aws-parallelcluster-platform" "aws-parallelcluster-environment" "aws-parallelcluster-computefleet" "aws-parallelcluster-shared" "aws-parallelcluster-tests")
+COOKBOOKS=("aws-parallelcluster-awsbatch" "aws-parallelcluster-config" "aws-parallelcluster-entrypoints" "aws-parallelcluster-scheduler-plugin" "aws-parallelcluster-slurm" "aws-parallelcluster-test" "aws-parallelcluster-platform" "aws-parallelcluster-environment" "aws-parallelcluster-computefleet" "aws-parallelcluster-shared" "aws-parallelcluster-tests")
```
Comment (awslitvit) [inline util/bump-version.sh]: Nitpick: we should automate cookbook list retrieval... But I really hope we won't change it in a foreseeable future, so it's ok to keep it this way.

The same goes for metadata files.

File: util/bump-version.sh [hunk 5]
```diff
@@ -29,25 +29,24 @@ sed -i "s/version '${CURRENT_PCLUSTER_VERSION_SHORT}'/version '${NEW_PCLUSTER_VE
...
 sed -i "s/depends 'aws-parallelcluster-test', '~> ${CURRENT_PCLUSTER_VERSION_SHORT}'/depends 'aws-parallelcluster-test', '~> ${NEW_PCLUSTER_VERSION_SHORT}'/g" ${FILES[*]}
 
-CURRENT_AWSBATCH_CLI_VERSION=$(sed -ne "s/^default\['cluster'\]\['parallelcluster-awsbatch-cli-version'\] = '\(.*\)'/\1/p" cookbooks/aws-parallelcluster-common/attributes/common.rb)
+CURRENT_AWSBATCH_CLI_VERSION=$(sed -ne "s/^default\['cluster'\]\['parallelcluster-awsbatch-cli-version'\] = '\(.*\)'/\1/p" cookbooks/aws-parallelcluster-shared/attributes/versions.rb)
 
-sed -i "s/default\['cluster'\]\['parallelcluster-awsbatch-cli-version'\] = '${CURRENT_AWSBATCH_CLI_VERSION}'/default['cluster']['parallelcluster-awsbatch-cli-version'] = '${NEW_AWSBATCH_CLI_VERSION}'/g" cookbooks/aws-parallelcluster-common/attributes/common.rb
+sed -i "s/default\['cluster'\]\['parallelcluster-awsbatch-cli-version'\] = '${CURRENT_AWSBATCH_CLI_VERSION}'/default['cluster']['parallelcluster-awsbatch-cli-version'] = '${NEW_AWSBATCH_CLI_VERSION}'/g" cookbooks/aws-parallelcluster-shared/attributes/versions.rb
```
Comment (awslitvit) [inline util/bump-version.sh]: I'd suggest to test this script if you didn't do it yet.

**Review by awslitvit — DISMISSED** [review]

File: .github/workflows/dokken-unit-tests.yml (hunk 1 above)
Comment (enrico-usai) [reply .github/workflows/dokken-unit-tests.yml:3]: yeah I think we can remove it but I'll do it in a separate patch.

File: cookbooks/aws-parallelcluster-computefleet/test/inspec.yml (hunk 2 above)
Comment (enrico-usai) [reply cookbooks/aws-parallelcluster-computefleet/test/inspec.yml:7]: I cannot find any reference of `aws-parallelcluster-common` in the CLI code.

File: util/bump-version.sh (hunk 5 above)
Comment (enrico-usai) [reply util/bump-version.sh]: Tested and improved, thanks for the callout!

**Review by awslitvit — APPROVED** [review]

**Pull request merged**